x-pack/plugins/ml/public/application/components/navigation_menu/date_picker_wrapper/date_picker_wrapper.tsx

```diff
@@ -88,6 +101,67 @@ export const DatePickerWrapper: FC = () => {
     timefilter.isTimeRangeSelectorEnabled()
   );
 
+  const refreshInterval = useMemo((): RefreshInterval => {
+    const resultInterval = globalState?.refreshInterval ?? timeFilterRefreshInterval;
+
+    /**
+     * Enforce pause when it's set to false with 0 refresh interval.
+     */
+    const pause = resultInterval.pause || (!resultInterval.pause && resultInterval.value <= 0);
+    const value = resultInterval.value;
+
+    return { value, pause };
+  }, [JSON.stringify(globalState?.refreshInterval), timeFilterRefreshInterval]);
+
+  useEffect(
+    function warnAboutShortRefreshInterval() {
+      const isResolvedFromUrlState = !!globalState?.refreshInterval;
+      const isTooShort = refreshInterval.value < DEFAULT_REFRESH_INTERVAL_MS;
+
+      // Only warn about short interval with enabled auto-refresh.
+      if (!isTooShort || refreshInterval.pause) return;
+
+      displayWarningToast(
+        {
+          title: isResolvedFromUrlState
+            ? i18n.translate('xpack.ml.datePicker.shortRefreshIntervalURLWarningMessage', {
+                defaultMessage:
+                  'The refresh interval in the URL is shorter than the minimum supported by Machine Learning.',
+              })
+            : i18n.translate('xpack.ml.datePicker.shortRefreshIntervalTimeFilterWarningMessage', {
+                defaultMessage:
+                  'The refresh interval in Advanced Settings is shorter than the minimum supported by Machine Learning.',
+              }),
+          text: toMountPoint(
+            wrapWithTheme(
+              <EuiButton
+                onClick={setRefreshInterval.bind(null, {
+                  pause: refreshInterval.pause,
+                  value: DEFAULT_REFRESH_INTERVAL_MS,
+                })}
+              >
+                <FormattedMessage
+                  id="xpack.ml.pageRefreshResetButton"
+                  defaultMessage="Reset to {defaultInterval}"
```

Comment:
I'd edit this `Set to {defaultInterval}` as 'Reset' implies to me that it was always `10s` before